agent/accumulator.go

```diff
@@ -161,6 +164,14 @@ func (ac *accumulator) AddFields(
 	ac.metrics <- m
 }
 
+func (ac *accumulator) AddError(err error) {
+	if err == nil {
+		return
+	}
+	atomic.AddUint64(&ac.errCount, 1)
+	log.Printf("ERROR in input [%s]: %s", ac.inputConfig.Name, err)
```

Comment:
yes but this function is reading it, so unless you make the function a value receiver, there can be a race condition here.